workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,685 @@
+
+- `git add --all`
+- `git commit -m "Geometric Pattern Workshop"`
+- `git push`
+
+Then, enter your GitHub username and password (careful here, as password will not be displayed).
+
+Congratulations! Your pattern is now live on `USERNAME.github.io/geometric_pattern/` (replace `USERNAME` with your own GitHub username!)
+
+## Part VI: Animate It
+
+WARNING: THIS SECTION IS NOT RECOMMENDED FOR THOSE WITH A HISTORY OF EPILEPSY. Even if you have not been previously diagnosed with epilepsy, you should be careful, and immediately close the window and consult a doctor if you experience any of the following symptoms: lightheadedness, altered vision, eye or face twitching, jerking or shaking of arms or legs, disorientation, confusion, or momentary loss of awareness.
+
+Let's first add the line to change the `draw()` function from running 60 times a second to 5 times a second. We can do this by using the `frameRate()` function in `setup()`:
+
+```js
+function setup() {
+  createCanvas(480,600);
+
+  frameRate(5);
+
+  circleDiameter = width/NUM_CIRCLES;
+  circleRadius = circleDiameter/2;
+}
+```
+
+That said, we can create a cool scrolling color effect by manipulating our colors to cycle. Right now we are resetting our initial RGB values each time `draw()` runs. But if we didn't reset every time, we could have rotating colors.
+
+```js
+function setup() {
+  createCanvas(480,600);
+
+  frameRate(5);
+
+  circleDiameter = width/NUM_CIRCLES;
+  circleRadius = circleDiameter/2;
+  rVal = 255;
+  gVal = 0;
+  bVal = 0;
+}
+
+function draw() {
+  var alternatingRow = false;
+
+...
+```
+
+If we move the three lines that set the initial RGB values to `setup()`, then we will be increment the values indefinitely. If we save and refresh our Live Preview, we'll see that everything is white. What's the deal?
+
+Since the max values of RGB are 255, 255, 255 (which makes white), any values above that will be white. The `draw()` function executes so many times per second that you can't see the color change progression.
+
+We can have color progression while keeping the values below 255 by [modding](https://en.wikipedia.org/wiki/Modulo_operation) each of these by 256. "Modding a by b" means that we divide a by b, and take the remainder. For example, 5mod2 gives us 1, because 5 divided by 2 has a remainder of 1. This arithmetic operation is available to us in JavaScript with the modulo (`%`) operator.
+
+Let's change our incrementation code to incorporate this:
+
+```js
+rVal = (rVal - 2)%256;
+gVal = (gVal + 7)%256;
+bVal = (bVal + 3)%256;
+```
+
+JavaScript `%` operator does something stupid, in that it mods negative numbers incorrectly. We can get around this by recognizing that subtracting 2 and modding is the same as adding 254 and modding.
+
+```js
+rVal = (rVal + 254)%256;
+```
+
+Save and refresh and be warned that it might be jarring.
+
+As a check, your `main.js` should look like this:
+
+Final code:
+
+```js
+var NUM_CIRCLES = 12;
+var circleDiameter;
```

Comment:
I'd put a newline after 618 just to create visual separate between constants and regular variables.